workflow/snakemake_rules/remote_files.smk

```diff
@@ -100,8 +100,15 @@ def path_or_url(path_or_url, stay_on_remote = False, keep_local = False):
     # fragment (including their leading delimiters) if they're empty.
     schemeless_url = urljoin(url.netloc + url.path, f"?{url.query}#{url.fragment}")
 
-    return SCHEME_REMOTES[url.scheme]()(
+    url = SCHEME_REMOTES[url.scheme]()(
```

Comment:
I might avoid reassigning `url` here, but not a big deal.

Reply:
Yeah, I balked at this a bit, couldn't think of anything more appropriate, and am ok with it now.